Mlem/Views/Tabs/Search/Search View.swift

```diff
@@ -0,0 +1,183 @@
+            content
+                .handleLemmyViews()
+                .navigationBarTitleDisplayMode(.inline)
+                .navigationTitle("Search")
+        }
+        .searchable(text: getSearchTextBinding(), prompt: "Search for communities")
+        .onSubmit(of: .search) {
+            preformSearch()
+        }
+    }
+    
+    @ViewBuilder
+    private var content: some View {
+        if showRecentSearches {
+            recentSearches
+        } else {
+            searchedContents
+        }
+    }
+    
+    @ViewBuilder
+    private var recentSearches: some View {
+        List {
+            Section {
+                ForEach(recentSearchesTracker.recentSearches, id: \.self) { recentlySearchedText in
+                    Button(recentlySearchedText) {
+                        searchText = recentlySearchedText
+                        preformSearch()
+                    }
+                }
+            } header: {
+                Text(recentSearchesTracker.recentSearches.isEmpty ? "No recent searches" : "Recent searches")
+            }
+            
+            Button(role: .destructive) {
+                recentSearchesTracker.clearRecentSearches()
+            } label: {
+                Label("Clear recent searches", systemImage: "trash").foregroundColor(.red)
+            }
+        }.listStyle(.insetGrouped)
+    }
+    
+    @ViewBuilder
+    private var searchedContents: some View {
+        if isSearching && communitySearchResultsTracker.foundCommunities.isEmpty {
+            LoadingView(whatIsLoading: .search)
+        } else if communitySearchResultsTracker.foundCommunities.isEmpty {
+            Text("No communities found for search")
+        } else {
+            List {
+                ForEach(communitySearchResultsTracker.foundCommunities) { community in
+                    CommunityLinkView(
+                        community: community.community,
+                        extraText: "\(community.counts.subscribers.roundedWithAbbreviations) subscribers"
+                    )
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                        .onAppear {
+                            let communityIndex = communitySearchResultsTracker.foundCommunities.firstIndex(of: community)
+                            if let index = communityIndex {
+                                
+                                // If we are half a page from the end, ask for more
+                                let distanceFromEnd = communitySearchResultsTracker.foundCommunities.count - index
+                                if distanceFromEnd == searchPageSize / 2 {
+                                    if hasMorePages {
+                                        preformSearch()
+                                    }
+                                }
+                            }
+                        }
+                }
+            }
+        }
+    }
```

Comment:
Good for now, long-term we can look at maybe using one of those shiny generic feed trackers